tests/test_mongodb_log_based_interruptible.py

```diff
@@ -0,0 +1,330 @@
+
+        # _id of the last record sync'd for each stream is the bookmarked
+        # max_id_value from the interrupted_state passed to the tap
+        self.assertEqual(records_by_stream['simple_coll_2']['messages'][-1]['data']['_id'],
+                         interrupted_state['bookmarks']['simple_db-simple_coll_2']['max_id_value'])
+
+        # verify the second sync bookmarks matches our expectatinos
+        for tap_stream_id in initial_state['bookmarks'].keys():
+
+            # verify the bookmark keys have not changed in a resumed sync
+            self.assertSetEqual(
+                set(initial_state['bookmarks'][tap_stream_id].keys()), set(second_state['bookmarks'][tap_stream_id].keys())
+            )
+
+            # check the table versions is the same
+            self.assertEqual(second_state['bookmarks'][tap_stream_id]['version'], initial_state['bookmarks'][tap_stream_id]['version'])
+
+            # verify the method has not changed
+            self.assertEqual(initial_state['bookmarks'][tap_stream_id]['last_replication_method'],
+                             second_state['bookmarks'][tap_stream_id]['last_replication_method'])
+
+            # verify the resumed sync has completed
+            self.assertTrue(second_state['bookmarks'][tap_stream_id]['initial_full_table_complete'])
+
+            # for the previously synced tables verify the oplog_ts_time moved forward
+            self.assertGreater(second_state['bookmarks'][tap_stream_id]['oplog_ts_time'],
+                               initial_state['bookmarks'][tap_stream_id]['oplog_ts_time'])
+
+        # verify the currently syncing state is none
+        self.assertIsNone(second_state['currently_syncing'])
+
+        #### Run 3rd Sync #####
+
+        third_sync = runner.run_sync_mode(self, conn_id)
+        third_sync_exit_status = menagerie.get_exit_status(conn_id, third_sync)
+        menagerie.verify_sync_exit_status(self, third_sync_exit_status, third_sync)
+
+        records_by_stream_3 = runner.get_records_from_target_output()
+
+        record_count_by_stream_3 = runner.examine_target_output_file(self,
+                                                                     conn_id,
+                                                                     self.expected_sync_streams(),
+                                                                     self.expected_pks())
+
+
+        # Verify the record count for the 3rd sync
+        for tap_stream_id in self.expected_sync_streams():
+            self.assertEqual(record_count_by_stream_3[tap_stream_id], self.expected_row_count_3()[tap_stream_id])
+
+        # verify the only record replicated is the last inserted record in the collection 2
+        self.assertEqual(len(records_by_stream_3['simple_coll_2']['messages']), 2)
+        self.assertEqual(records_by_stream_3['simple_coll_2']['messages'][0]['action'], 'activate_version')
```

Comment:
I am forgetting, what is the purpose of this activate_version message? Is it expected?

Reply:
Yes it is expected and i am validating the presence of activate message and the actual document. Since the length of the message is 2, we have to verify if the other message is activate version